src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h

```diff
@@ -27,6 +27,21 @@
 #include "src/core/lib/gprpp/ref_counted.h"
 #include "src/core/lib/security/security_connector/ssl_utils.h"
 
+struct grpc_tls_err_details
+    : public grpc_core::RefCounted<grpc_tls_err_details> {
+ public:
+  grpc_tls_err_details() : error_details_("") {}
+  void set_err_details(const char* err_details) {
+    error_details_ = err_details;
+  }
+  const std::string& err_details() { return error_details_; }
+  const char* err_details_c_str() { return error_details_.c_str(); }
+  bool has_error() { return !error_details_.empty(); }
```

Comment:
Is the API necessary? Can we call the getter and then call `empty()`?

Reply:
Done.